service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/CreateAzureStorageContainerStep.java

```diff
@@ -0,0 +1,142 @@
+
+  public CreateAzureStorageContainerStep(
+      AzureConfiguration azureConfig,
+      CrlService crlService,
+      ControlledAzureStorageContainerResource resource) {
+    this.azureConfig = azureConfig;
+    this.crlService = crlService;
+    this.resource = resource;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    final AzureCloudContext azureCloudContext =
+        context
+            .getWorkingMap()
+            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+    final StorageManager storageManager = crlService.getStorageManager(azureCloudContext, azureConfig);
+
+    try {
+      storageManager
+          .blobContainers()
+          .defineContainer(resource.getStorageContainerName())
+          .withExistingStorageAccount(azureCloudContext.getAzureResourceGroupId(), resource.getStorageAccountName())
+          .withPublicAccess(PublicAccess.NONE)
+          .withMetadata("workspaceId", resource.getWorkspaceId().toString())
+          .withMetadata("resourceId", resource.getResourceId().toString()).
+              create(
+                Defaults.buildContext(
+                  CreateStorageContainerRequestData.builder()
+                      .setName(resource.getStorageContainerName())
+                      .setStorageAccountName(resource.getStorageAccountName())
+                      .setResourceGroupName(azureCloudContext.getAzureResourceGroupId())
+                      .build()));
+
+    } catch (ManagementException e) {
+      logger.error(
+          "Failed to create the Azure storage container '{}' with storage account with the name '{}'. Error Code: {}",
+          resource.getStorageContainerName(),
+          resource.getStorageAccountName(),
+          e.getValue().getCode(),
+          e);
+
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
+    }
+
+    return StepResult.getStepResultSuccess();
+  }
+
+  /**
+   * Deletes the storage container if the container is available. If the storage container is available
+   * and deletes fails, the failure is considered fatal and must looked into it.
+   *
+   * @param context
+   * @return Step result.
+   * @throws InterruptedException
+   */
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    final AzureCloudContext azureCloudContext =
+            context
+                    .getWorkingMap()
+                    .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+    final StorageManager storageManager = crlService.getStorageManager(azureCloudContext, azureConfig);
+
+    try {
+      storageManager
+              .storageAccounts()
+              .getByResourceGroup(
+                      azureCloudContext.getAzureResourceGroupId(), resource.getStorageAccountName());
+    } catch (ManagementException ex) {
+      logger.warn(
```

Comment:
Is it certain that all `ManagementException`s thrown in the above statement will be of the not found variety? It seems like there may be other types of errors that this may be masking.